flink-core/src/main/java/org/apache/flink/api/common/ExecutorFactory.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.common;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.configuration.Configuration;
+
+import java.net.URL;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * A ExecutorFactory create the specific implementation of {@link Executor}
+ * 
+ */
+@Internal
+public class ExecutorFactory<T extends Executor> {
+
+	private static final String LOCAL_EXECUTOR_CLASS = "org.apache.flink.client.LocalExecutor";
+	private static final String REMOTE_EXECUTOR_CLASS = "org.apache.flink.client.RemoteExecutor";
+
+	private Class<T> clazz;
+
+	public ExecutorFactory(Class<T> clazz) {
+		this.clazz = clazz;
+	}
+
+	// ------------------------------------------------------------------------
+	//  Executor Factories
+	// ------------------------------------------------------------------------
+	
+	/**
+	 * Creates an executor that runs locally in a multi-threaded environment.
+	 * 
+	 * @return A local executor.
+	 */
+	public T createLocalExecutor(Configuration configuration) {
```

Comment:
Can we make this a static method? Then we would not have to create a factory object.